### DISCUSSION TITLE
Правки миров // zen

### DISCUSSION THREAD

**Discussion by dosinabox** [dosinabox/g2nr_unofficial_update#445] [General]
Здесь будут перечисляться все найденные косяки миров, а также предложения по улучшению.

**Comment by dosinabox** [comment]
Хоринис (NEWWORLD.ZEN):

- [x] отключенные шипы в тайнике воров срабатывают, если изучена акробатика
- [x] щит в кузнице казармы висит в воздухе
- [ ] сделать новую модель больших щитов
- [ ] повернуть точки для предводителей орков у Ксардаса и около пещеры бандитов Малета
- [ ] сделать кровати в тюрьме, можно взять новую модель коек с цепями
- [ ] модель STARGATE_EDGES.3DS не существует
- [ ] проницаемые книжные стенды
- [x] стенд в казармах без текста - придумать новый или использовать существующий
- [x] стенд в казармах без подписи
- [x] изучить ошибки анализатора (повторные имена точек, пустые объекты и т.д.)
- [x] тарелка в доме Саландрила торчит в шкафу
- [x] за скалой в порту около корабля паладинов очень много объектов
- [ ] открытая решетка тайной комнаты Ксардаса торчит через башню
- [x] исправить все BOOK_BLUE.ASC на динамическую модель, чтобы листались страницы
- [x] шкафы у Неораса торчат друг в друге
- [x] подвинуть точку с ключом в таверне Орлана ближе к краю шкафа или вставлять сам ключ
- [x] заполнить комнату Онара редкими или красивыми предметами, о которых говорит Василий
- [x] заменить картины в тайнике воров на новые уникальные модели
- [x] добавить сундук Сифера и кровать для Бодо
- [x] повернуть паладинов около Альбрехта и Хагена
- [x] сундуки бандитов около Элвриха висят в воздухе
- [x] задействовать рудные жилы
- [x] что-то на полу в доме Эдды и комнате в борделе с Пеком
- [x] всем фонарям города добавить требование факела для зажигания
- [x] проницаемая статуя орка в пещере около города
- [x] незапертый сундук на первом этаже маяка Джека с замком
- [x] незапертый сундук в тайнике воров с замком
- [x] три травы под текстурами у лагеря бандитов за фермой Секоба
- [x] два серафиса у палатки на месте раскопок магов воды
- [x] лечебная трава в долине Лестера (около пещеры с крысами в камнях)
- [x] неправильные названия надгробных плит (см. [комментарий](https://github.com/dosinabox/g2nr_unofficial_update/issues/49#issuecomment-626289878))
- [x] проницаемые шкафы в монастыре
- [x] точка для спауна моллюска за текстурой в скале в гавани
- [x] сундукам на ферме Онара прописана неправильная гильдия GIL_GRD
- [x] царский щавель над пещерой мракориса у маяка немного торчит в земле
- [x] сундук в маяке Джека - починить комбинацию или оставить незапертым как в оригинале?
- [x] при игре с DX11 в пещере с Грозой драконов торчат деревья, нужно поднять их над пещерой
- [x] autoCamPlayerMovable=0 для камеры у портала в Яркендар
- [x] добавить триггер в пещере со спящими големами, чтобы их можно было пробуждать

**Comment by dosinabox** [comment]
Миненталь (OLDWORLD.ZEN):

- [x] стенды в старой башне Ксардаса
- [x] повернуть сундук около пещеры орков за мостом
- [x] WAY: Ladder highpoint error OC_GUARD_FLOOR_UP_02/OC_GUARD_FLOOR_UP_03
- [x] убрать сыр из сундука около Талбина
- [x] задействовать рудные жилы
- [x] vobFarClipZScale=float:20901890
- [x] заменить руну телепорта к проходу на точку для вставки предмета
- [x] добавить точки FP_ROAM в местах обитания мясных жуков
- [x] переименовать FP_GUARD_OC_THRONE_0* в FP_STAND_OC_THRONE_0*
- [x] добавить точки для отдыха каторжников-рабочих на стене
- [x] привязать двери в запретные комнаты к GIL_PAL
- [x] добавить источники освещения к факелам у входа в тюрьму

**Comment by dosinabox** [comment]
Яркендар (ADDONWORLD.ZEN):

- [x] в сундуках Скипа и за домом Фиска - несуществующий предмет itfo_meat (заменить на itfomutton)
- [x] щель в земле около гробницы Куарходрона (закрыта камнем, но лучше поправить меш)
- [x] Марус застревает в котле, когда отходит помочиться
- [x] кривая баночка itpo_littlemana в библиотеке
- [x] добавить служебный сундук для отслеживания версий D36_SECRET_DEV_CHEST
- [x] удалить руну огненной стрелы и добавить точку для вставки другого предмета (FP_ITEM_SWAMP_RUNE_01)
- [x] добавить тексты пустым скрижалям (Use_BookstandMayaCommon_01)
- [x] повторные FP_STAND_MINEGUARD_01 и FP_STAND_MINEGUARD_02
- [x] добавить точки для рабов в домах у золотой шахты
- [x] табак в доме Грега
- [x] приподнять точки для вставки орков у библиотеки
- [x] повернуть и чуть подвинуть BL_DOWN_CENTER_07_B
- [x] заменить стул Скиннера на обычную модель или модель черепа на предмет
- [x] реализовать падение шатающегося дерева
- [x] разъединить лишнюю связь между точками у башни бандитов и на пляже с огненными ящерами
- [x] приподнять точку FP_ADW_GREGSBOTTLE
- [x] незапертый сундук с замком около ADW_ADANOSTEMPEL_TREASUREPITS_06
- [x] неправильный материал меша в пещере с точкой ADW_CANYON_PATH_TO_BANDITS_14

**Comment by dosinabox** [comment]
> Добрый день. Все функции, которые вызываются в Mobsi объектах надо в скриптах прописать заглавными буквами.
> Недавно делал функцию для двери, и пока не написал заглавными буквами, код не работал:confused:
> Например книжные стенды, или triggerscript
> Для простоты, сохрани миры не в скомпилированном варианте.
> В notepad++ делаешь поиск по этим мирам написав scriptFunc=string:
> Сразу будет видно, какие скрипты надо прописать большой буквой.
> Также onStateFunc=string:
> В некоторых mobsi прописано с маленькой буквы - надо исправить на большую.

**Comment by dosinabox** [comment]
Надгробные плиты:

Неиспользуемые:

- MOBNAME_GRAVE_01 "Барон Генрих фон Сталь 551-589 - 'Пришел, увидел и упал замертво'"; _//применено_
- MOBNAME_GRAVE_02 "Бертран 465-480 - 'Я всегда любил есть поганки'"; _//применено_
- MOBNAME_GRAVE_05 "Декс Кантионис 325-431 - 'Я уже несколько дней ничего не ел'"; _//применено_
- MOBNAME_GRAVE_06 "Утар Лайтбрингер 205-532"; _//применено_
- MOBNAME_GRAVE_08 "Онурб 634-579 - 'Дерепан модаз'"; _//применено_

У двух плит одинаковое название (в пещере в долине Лестера и на кладбище за таверной "Мертвая гарпия"):

- MOBNAME_GRAVE_18 "Иотар 721-762" (в пещере надо MOBNAME_GRAVE_08); _//исправлено_

![20200504190703_1](https://user-images.githubusercontent.com/39790055/81494098-aba67880-92ae-11ea-9d74-63a3b7794d0d.jpg)
![20200504190627_1](https://user-images.githubusercontent.com/39790055/81494097-aa754b80-92ae-11ea-8f16-4033c64a03b8.jpg)

Плиты без названий:

- 1 в Яркендаре (большая пещера с ползунами); _//применено MOBNAME_GRAVE_06_
- 1 на Ирдорате (комната рядом с Педро);
- 3 в восточном лесу Хориниса; _//применены MOBNAME_GRAVE_01, MOBNAME_GRAVE_02 и MOBNAME_GRAVE_05_
- 1 у прохода Хориниса (надо MOBNAME_GRAVE_09); _//исправлено_

![20200504204432_1](https://user-images.githubusercontent.com/39790055/81494086-9893a880-92ae-11ea-8fb9-40fd0fc839ba.jpg)
![20200504190147_1](https://user-images.githubusercontent.com/39790055/81494087-99c4d580-92ae-11ea-9999-e894075d63d1.jpg)

**Comment by dosinabox** [comment]
> * 'Дерепан модаз'

Для справки: изначальный перевод Акеллы 'Этлемеркегму красный', но тут всё проще, если посмотреть другие языковые версии и учесть, что русский перевод производился с английского:
'Этлемеркегму красный' (рус.) = 'Etlepmerkegmu red' (англ.) = 'Etlepmerkegmu red' (нем.)

'Etlepmerkegmu red' задом наперед это 'Der umgekrempelte', что в переводе с немецкого значит 'задом наперед' / 'вверх ногами'. Теперь понятно, почему первая дата на плите больше второй. А Онурб - это Бруно.

В общем, при переводе на английский с адаптацией не заморочились, а на русский просто перевели в лоб. А вот поляки молодцы: 'Dózrp an łyt' - 'Tył na przód'.

Но, впрочем, эта надпись в игре всё равно не используется.

**Comment by Kor-Angar** [comment]
в минентале в разрушенной башни ксардаса если сундук в воде

**Reply by Kor-Angar** [reply]
герой там периодически подвисает после "выхода" из сундука

хотя может это только потому что я из сундука выхожу через вытаскивание оружия

**Reply by Kor-Angar** [reply]
> Именно поэтому. И передвигать этот сундук не хочется, он там удачно спрятан)

**Comment by dosinabox** [comment]
![39510_screenshots_20210202202240_1](https://user-images.githubusercontent.com/39790055/106922884-8bf98f00-671e-11eb-9e63-9f51264b383f.jpg)

**Comment by dosinabox** [comment]
autoCamPlayerMovable = 0 (FALSE)
autoCamPlayerMovable - can the Player move around while the camera is running
1 / TRUE - The player will be able to move
0 / FALSE - The player is motionless

**Comment by dosinabox** [comment]
Поиграв в Архолос, появилась мысль на будущее: сделать канализацию Хориниса отдельной локацией с точками загрузки в порту и в подвале гостиницы. Это как минимум улучшит производительность в городе. Что думаешь?

**Reply by Kor-Angar** [reply]
она не такая большая и транзитная. т.е. герой заходит в канализацию (взять квест), город (на квест), канализация(сдать квест). в самой канализации квестов нет и она маленькая. еще в канализации можно поучиться, но получится такая штука что я забегаю туда загрузка мира, бегу к Кассии тренировать ловкость и выбегаю, опять загрузка.

можно храм в котором равен сделать отдельным миром. от этого будет толк так в храм герой идёт единовременно и проводит там много времени. плюс тогда освещение мира не будет влиять(?) на освещение в храме и ночью и днем там будет одинакова темно.(или нет?).

**Reply by Kor-Angar** [reply]
> Это как минимум улучшит производительность в городе.

Это главная задумка, потому что место самое нагруженное в игре. А если не улучшит, то и смысла нет.
Из других плюсов: никого нельзя будет выманить на улицу (и наоборот), запрет на проникновение превращением через потолок, исправление реакций граждан через текстуры (сейчас костыль в восприятиях), ускорение загрузки сохранений в Хоринисе (незначительное).

**Reply by Kor-Angar** [reply]
не знаю, что тут сказать :-) я понимаю эти плюсы и если бы канализация была полноценной локацией в которое герой проводил хоть какое то время, а не забегал туда на одну минуты я бы был двумя руками за. но учитывая что канализация забежал/убежал и это длится буквально пару минут, то не могу сказать точно да.

но можно думать глобальнее, а если попытаться отделить весь город Хоринос? :-)

**Comment by dosinabox** [comment]
> плюс тогда освещение мира не будет влиять(?) на освещение в храме и ночью и днем там будет одинакова темно.(или нет?).

Там всё нормально. У самого входа начинается зона с фиксированным освещением:
![Screenshot_1](https://user-images.githubusercontent.com/39790055/148441961-505af272-2d65-4df9-b75e-04c2aee41dcf.jpg)

**Comment by Kor-Angar** [comment]
сундукам можно дать уникальное имена, что бы можно было в случае необходимости делать проверки на уровне скриптов и менять лут:

- [x] сундук Игарацио
- [x] сундук Дарона (в доме Зуриуса)
- [x] сундук в пещере для испытания огня // MAGICCHEST
- [ ]

**Reply by dosinabox** [reply]
> сундук в пещере для испытания огня

MAGICCHEST

**Reply by Kor-Angar** [reply]
в 3 главе можно в MAGICCHEST добавить что нибудь при игре за Пала / ОнД. а ключ закинуть например в какого нибудь ищущего.

**Reply by Kor-Angar** [reply]
по моему только заглавными надо писать буквами? 

			vobName=string:IGARAZCHEST


```D
[% oCMobContainer:oCMobInter:oCMOB:zCVob 64513 7618]
			pack=int:0
			presetName=string:
			bbox3DWS=rawFloat:47953.6445 4992.00098 21682.1895 48058.2148 5069.46875 21787.4258 
			trafoOSToWSRot=raw:f08a30bf00000000436339bf000000000000803f000000004463393f00000000f38a30bf
			trafoOSToWSPos=vec3:48007.6992 4992 21736.9063
			vobName=string:IGARAZCHEST
			visual=string:CHESTBIG_NW_NORMAL_LOCKED.MDS
			showVisual=bool:1
			visualCamAlign=enum:0
			visualAniMode=enum:0
			visualAniModeStrength=float:0
			vobFarClipZScale=float:1
			cdStatic=bool:1
			cdDyn=bool:1
			staticVob=bool:1
			dynShadow=enum:0
			zbias=int:0
			isAmbient=bool:0
			[visual zCModel 0 7619]
			[]
			[ai % 0 0]
			[]
			focusName=string:MOBNAME_IGARAZ
			hitpoints=int:10
			damage=int:0
			moveable=bool:0
			takeable=bool:0
			focusOverride=bool:0
			soundMaterial=enum:0
			visualDestroyed=string:
			owner=string:NOV_601_Igaraz
			ownerGuild=string:
			isDestroyed=bool:0
			stateNum=int:1
			triggerTarget=string:
			useWithItem=string:
			conditionFunc=string:
			onStateFunc=string:
			rewind=bool:0
			locked=bool:-1
			keyInstance=string:ItKe_IgarazChest_mis
			pickLockStr=string:
			contains=string:ItWr_BabosDocs_MIS
		[]
```

**Reply by dosinabox** [reply]
желательно, но не обязательно

**Reply by Kor-Angar** [reply]
а как предмет из сундука удалить? :-)  фунцкия вставки есть, а удаление что то не нашёл..
 
```D
func void B_ENTER_NEWWORLD_Kapitel_1()
{
	B_ENTER_NEWWORLD_Kapitel_1_old();
	Mob_CreateItems ("IGARAZCHEST",ITAR_Thorus_Addon,13);
//	Mob_HasItems
};

```

**Reply by dosinabox** [reply]
> а как предмет из сундука удалить?

https://worldofplayers.ru/threads/13198/post-1151170

**Reply by Kor-Angar** [reply]
**zParserExtender** 

```D
 // Удаляет из контейнера моба object предмет item
 func void Mob_RemoveItem( var instance object , var int item )

 // Удаляет из контейнера моба object предметы item в количестве cnt
 func void Mob_RemoveItems( var instance object , var int item , var int cnt )
```

надо будет попробовать :-)

**Reply by Kor-Angar** [reply]
не хочет. на синтаксис ругается


```D
func void B_ENTER_NEWWORLD_Kapitel_1()
{
	B_ENTER_NEWWORLD_Kapitel_1_old();
	
	Mob_CreateItems ("IGARAZCHEST",ITAR_Thorus_Addon,1);
	//Mob_RemoveItems ("IGARAZCHEST",ItWr_BabosDocs_MIS,1);
	Mob_RemoveItem ("IGARAZCHEST",ItWr_BabosDocs_MIS);
};
```

**Reply by dosinabox** [reply]
Пожалуйста, обнови список сундуков, кому какое имя дать. В ближайшее время займусь мирами.

**Reply by Kor-Angar** [reply]
все не вспомнить. ну вот сундук что под мостом возле таверны мёртвая гарпия. его можно будет задействовать при игре не магом.

**Reply by Kor-Angar** [reply]
сундук в склепе в минентале. где амулет Ангара

**Reply by dosinabox** [reply]
добавлено:
CHEST_DARON - сундук Дарона
CHEST_IGARAZ - сундук Игараца
CHEST_ANDRE - сундук Андрэ
CHEST_ORLAN - сундук Орлана
CHEST_DEXTER - сундук Декстера
CHEST_LARIUS - сундук Лариуса
CHEST_BRIDGE_01 и CHEST_BRIDGE_02 - сундуки под мостом у таверны

**Reply by dosinabox** [reply]
CHEST_CAPTAIN_DI - сундук в каюте капитана на Ирдорате
CHEST_BLACKNOV - сундук в комнате послушника черного мага на Ирдорате
CHEST_FIREDRAGON_DI - сундук рядом с Феодароном

**Reply by dosinabox** [reply]
CHEST_CAPTAIN_NW - сундук в каюте капитана в Хоринисе
CHEST_KDFPLAYER - сундук в комнате ГГ-мага в монастыре
CHEST_CIPHER - сундук Сифера

**Reply by dosinabox** [reply]
CHEST_ONAR - сундук в спальне Онара
CHEST_WASILI - сундук в зале с Василием

**Reply by dosinabox** [reply]
Яркендар:
CHEST_KDW_START - сундук магов Воды
CHEST_FORTUNO - сундук Фортуно
CHEST_ESTEBAN - сундук Эстебана
CHEST_THORUS - сундук Торуса

**Reply by Kor-Angar** [reply]
CHEST_KDW_START - сундук магов Воды это в ярике?

**Reply by Kor-Angar** [reply]
может в название сразу отмечать мир? 

CHEST_AW_THORUS
CHEST_NW_CIPHER

**Reply by dosinabox** [reply]
Миненталь:
CHEST_CRYPT_01, CHEST_CRYPT_02 - сундуки в пещере скелета-мага
CHEST_XARDAS_01, CHEST_XARDAS_02 - сундуки в башне Ксардаса
CHEST_STORE - сундук на складе в замке
CHEST_GAROND - сундук Гаронда в его комнате
CHEST_SKELETON - сундук в башне замка, где скелет появляется
CHEST_CAVALORN - сундук в старой хижине Кавалорна

**Comment by dosinabox** [comment]
На "подумать" - как думаешь, зачем в подвале монастыря этот недопроход? Выглядит странно, никуда не ведет, ничего не содержит. Наверное, разработчики планировали секретный проход в этом месте, а потом перенести за шкаф и забыл это убрать. Можно будет удалить и сделать обычную стену.

![image](https://user-images.githubusercontent.com/39790055/170479316-679e0746-fecb-4c86-975b-16c10ac1ddd6.png)

**Reply by Kor-Angar** [reply]
а в готики 2 классик там так же?

**Reply by dosinabox** [reply]
да

**Reply by dosinabox** [reply]
По размерам сюда хорошо подходит дверь. Можно сделать за ней какую-нибудь кладовку с сундуком или полками и где-нибудь спрятать ключ.

**Reply by Kor-Angar** [reply]
тогда проще сделать открывающиеся стенку через светильник.

**Reply by Kor-Angar** [reply]
и скелет туда положить :-)

**Reply by dosinabox** [reply]
не проще)

**Reply by Kor-Angar** [reply]
закрой дырку шкафов и забей :-)

**Reply by dosinabox** [reply]
![Screenshot_3](https://user-images.githubusercontent.com/39790055/230367636-f458a32f-c994-46ec-a6ca-3737785bc780.jpg)

**Comment by dosinabox** [comment]
Неправильный материал в Яркендаре, возможно смешение текстур:
![mat3](https://github.com/dosinabox/g2nr_unofficial_update/assets/39790055/d1ccb208-9d1e-4323-89d8-97832788b6ed)

**Reply by dosinabox** [reply]
исправлено

**Comment by dosinabox** [comment]
Нужен список предметов, которые появились в OLDWORLD.ZEN в аддоне.

**Reply by Kor-Angar** [reply]
не понял вопроса.

**Reply by dosinabox** [reply]
чем отличается OLDWORLD.ZEN классики от аддона?

**Reply by Kor-Angar** [reply]
в классике больше травы
в аддоне добавлены скрижали

несущественные правки  в замке. где то доп кровать стоит.

**Reply by dosinabox** [reply]
а травы и зелья с бонусами?

**Reply by Kor-Angar** [reply]
да. ягоды гоблина, корень дракона, пару эликсиров

**Reply by dosinabox** [reply]
Вот такой список мне нужен) Где лежат предметы с бонусами в классике и в аддоне.

**Reply by Kor-Angar** [reply]
в ручную я такое не составлю. эти данные как то можно спарсить из мира. наверное если сохранить мир в декомпилированном виде.

а что ты в итоге хочешь? использовать один мир?

**Reply by Kor-Angar** [reply]
![image](https://github.com/dosinabox/g2nr_unofficial_update/assets/49514764/a8a8ac09-c839-4ff3-81ca-7327d69ab9c9)
 в немецкой базе есть интерактивная карта
http://mondgesaenge.de/G2DB/

**Reply by dosinabox** [reply]
> а что ты в итоге хочешь? использовать один мир?

Да, в перспективе. И пригодится для встраиваемого Wasteland. Можно проверять на классику/аддон и вставлять нужные предметы через скрипты.

**Reply by dosinabox** [reply]
Добавлено много точек в нужных местах, предметы можно вставлять из скриптов:
<img width="370" height="421" alt="image" src="https://github.com/user-attachments/assets/18c9e277-5587-4f2a-9dfb-57f975e50457" />
В Wasteland я вызываю `B_Wasteland_InsertClassicItems()`, если игрок использует классику:
https://github.com/dosinabox/gothic_plugins/blob/main/Wasteland/Seed_Wasteland_World.d

**Comment by dosinabox** [comment]
Сундуки Хориниса с пробелами в перечне содержимого:
```cpp
//дом Валентино:
contains=string: ITMI_Gold:50

//пещера с бандитами около города с топором орков:
contains=string:ItFo_SmellyFish,ItFo_Fish:3, ItMi_Gold:10

//сундук с зашитой рыбой Халвора в пещере с бандитами:
contains=string:ItFo_HalvorFish_MIS,ItFo_Fish:3, ItMi_Gold:25

//сундук Дарона:
contains=string:ItSc_Light,ItSc_Sleep:2,ITSC_Charm:2,ITSC_Icebolt, ItPo_Mana_01,ItMi_Gold:50
```
На фактическое наличие предметов не влияет, но лучше исправить при случае.

**Reply by dosinabox** [reply]
исправлено

**Comment by Kor-Angar** [comment]
@dosinabox  ты присваила книжным стендам уникальные текст? в частности в минентале.

**Reply by dosinabox** [reply]
Только пустым. Два в Минентале (Магия телепортации и Настоящая Сила) и один в Хоринисе (Дежурства).

**Reply by Kor-Angar** [reply]
тогда я пока так сделаю :-)

```D
func void Use_BookstandALCHEMY2_S1()
{
	if(C_NpcIsHero(self))
	{
    
        if Npc_GetDistToWP(hero,"OW_UNDEAD_DUNGEON_03") < 1000
        {
            Use_Manarezept_04();
        }
        else
        {
            Use_BookstandALCHEMY2_S1_old();
        };
    };
};


func void Use_BookstandAnimals2_S1()
{
	if(C_NpcIsHero(self))
	{
    
        if Npc_GetDistToWP(hero,"OW_UNDEAD_DUNGEON_03") < 1000
        {
            Use_Heilrezept_04();
        }
        else
        {
            Use_BookstandAnimals2_S1_old();
        };
    };
};
```
![20230914165524_1](https://github.com/dosinabox/g2nr_unofficial_update/assets/49514764/621c43a9-7a7c-4dfb-adaa-5c13df94a895)

**Reply by dosinabox** [reply]
А в чем задумка?

**Reply by Kor-Angar** [reply]
задумка простая. в таком знаковом месте какие то неинтересные тексты. а так герой узнает рецепты чистой маны и чистого здоровья.

**Reply by dosinabox** [reply]
Желательно ещё проверять текущий мир. Npc_GetDistToWP с несуществующей точкой возвращает число **2147483647**.

**Reply by Kor-Angar** [reply]
угу. тоже хотел но когда начал писать забыл. меня вообще эти чекины по wp всегда подгражают так как варианты есть разные и я так и не понял когда какой лучше использовать.

**Comment by dosinabox** [comment]
Нужно дать всем фонарям в Хоринисе уникальные имена и устанавливать им расписание с помощью **Wld_SetObjectRoutine** вместо **Wld_SetMobRoutine**. Это позволит контролировать каждый фонарь индивидуально и избавиться от этих предупреждений:
```
[w] 00:23 Warn:  0 C:       SCRIPT: Wld_SetMobRoutine(): could not find a vob with name: FIREPLACE .... <oGameExternal.cpp,#252>
[w] 00:23 Warn:  0 C:       SCRIPT: last parser func-name: INIT_NEWWORLD .... <oGameExternal.cpp,#262>
[w] 00:23 Warn:  0 C:       SCRIPT: Wld_SetMobRoutine(): could not find a vob with name: FIREPLACE .... <oGameExternal.cpp,#252>
[w] 00:23 Warn:  0 C:       SCRIPT: last parser func-name: INIT_NEWWORLD .... <oGameExternal.cpp,#262>
[w] 00:23 Warn:  0 C:       SCRIPT: Wld_SetMobRoutine(): could not find a vob with name: FIREPLACE .... <oGameExternal.cpp,#252>
[w] 00:23 Warn:  0 C:       SCRIPT: last parser func-name: INIT_NEWWORLD .... <oGameExternal.cpp,#262>
```

**Reply by Kor-Angar** [reply]
было бы хорошо давать им имена так что бы они соотносились с ближайшими WP

**Reply by dosinabox** [reply]
```cpp
func void INIT_SUB_NewWorld_Part_City_01()
{
//	Wld_SetMobRoutine(0,0,"FIREPLACE",1);
//	Wld_SetMobRoutine(20,0,"FIREPLACE",1);
//	Wld_SetMobRoutine(5,0,"FIREPLACE",0);
	Wld_SetObjectRoutine(0,0,"CITY_LAMP_01",1);
	Wld_SetObjectRoutine(20,0,"CITY_LAMP_01",1);
	Wld_SetObjectRoutine(5,0,"CITY_LAMP_01",0);
	...
```

**Reply by dosinabox** [reply]
Собственно, готово.
```cpp
B_SetObjectRoutine("CITYLAMP_01");
...
B_SetObjectRoutine("CITYLAMP_43");
```

**Comment by dosinabox** [comment]
Что если сделать точкой старта на Ирдорате не трап, а каюту капитана? И триггер начала шестой главы - на выходе из каюты? Плюсы:
- более плавный визуальный переход из Хориниса
- можно поговорить с капитаном и командой до схода на берег
- показывает игроку, что в обновлении каюта полностью доступна
- возможно, так и было задумано

**Reply by dosinabox** [reply]
> Я не был в каюте и не знал что в обновлении туда есть доступ :-)

В моде классики тоже есть.

**Comment by dosinabox** [comment]
![Screenshot_7](https://github.com/user-attachments/assets/3ff80b25-ec9d-4934-9021-475d3719be7e)

**Reply by dosinabox** [reply]
исправлено

**Comment by dosinabox** [comment]
Не закрыты порталами:

- [x] пещера Сагитты (Хоринис)
- [x] маяк Джека (Хоринис)
- [x] пещера с точкой ADW_ENTRANCE_RAKEPLACE_04 (Яркендар)

**Reply by dosinabox** [reply]
Получилось закрыть порталами все помещения горного форта в Минентале (там где каменный дракон).

**Comment by dosinabox** [comment]
Памятка по созданию порталов:
<img width="600" alt="image" src="https://github.com/user-attachments/assets/c8c7165e-7a38-4209-9e73-aff1985d625d" />
